src/lib/server/exec.c

```diff
@@ -826,6 +833,7 @@ int fr_exec_wait_start(REQUEST *request, fr_value_box_t *vb, VALUE_PAIR *env_pai
 				close(to_child[1]);
 			}
 			fr_strerror_printf("Failed opening pipe to read from child - %s", fr_strerror());
+			talloc_free(envp);
```

Comment:
When you've got a bunch of similar cleanup cases it's better just to add a goto and put an error label at the top of the function.

Reply:
It's true. let me update it.